compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1117,6 +1125,69 @@ public BLangNode transform(AnnotationNode annotation) {
         return bLAnnotationAttachment;
     }
 
+    @Override
+    public BLangNode transform(AnnotationDeclarationNode annotationDeclarationNode) {
+        BLangAnnotation annotationDecl = (BLangAnnotation) TreeBuilder.createAnnotationNode();
+        DiagnosticPos pos = getPosition(annotationDeclarationNode);
+        annotationDecl.pos = pos;
+        annotationDecl.name = createIdentifier(pos, annotationDeclarationNode.annotationTag().text());
+
+        if (annotationDeclarationNode.visibilityQualifier() != null) {
+            annotationDecl.addFlag(Flag.PUBLIC);
+        }
+
+        if (annotationDeclarationNode.constKeyword() != null) {
+            annotationDecl.addFlag(Flag.CONSTANT);
+        }
+
+        annotationDecl.annAttachments = applyAll(annotationDeclarationNode.metadata().annotations());
+
+        annotationDeclarationNode.metadata().documentationString().ifPresent(docString-> {
+            annotationDecl.markdownDocumentationAttachment = createMarkdownDocumentationAttachment(docString);
+        });
+
+        Node typedesc = annotationDeclarationNode.typeDescriptor();
+        if (typedesc != null) {
+            annotationDecl.typeNode = createTypeNode(typedesc);
+        }
+
+        SeparatedNodeList<Node> paramList = annotationDeclarationNode.attachPoints();
+
+        for (Node child : paramList) {
+            AnnotationAttachPointNode attachPoint = (AnnotationAttachPointNode) child;
+            boolean source = attachPoint.sourceKeyword() != null;
+            AttachPoint bLAttachPoint;
+            AttachPoint.Point firstIndent = AttachPoint.fromString(attachPoint.firstIdent().text().toLowerCase());
+            switch (firstIndent) {
+                case OBJECT:
+                    AttachPoint.Point secondIdent =
+                            AttachPoint.fromString(attachPoint.firstIdent().text().toLowerCase());
+                    switch (secondIdent) {
+                        case FUNCTION:
+                            bLAttachPoint = AttachPoint.getAttachmentPoint("objectfunction", source);
```

Comment:
can we replace the string using `org.ballerinalang.model.elements.AttachPoint` value ?

Reply:
Done